packages/odata/parsers.ts

```diff
@@ -91,6 +96,12 @@ export class BlobParser extends ODataParser<Blob> {
         r.blob().then(resolve);
     }
 }
+export class StreamParser extends ODataParser<IResponseBodyStream> {
+
+    protected parseImpl(r: Response, resolve: (value: any) => void): void {
+        resolve({ body: r.body, knownLength: parseInt(r.headers["content-length"], 10) });
```

Comment:
FYI: I'd gladly return just the stream to make the interface cleaner, but in many cases a stream without known length is not useful for sending further.